File: ObjectProgrammingJava/projekt po - 2/src/projekt/rosliny/Roslina.java
package projekt.rosliny;

import projekt.Organizm;
import projekt.Wypisanie;
import projekt.obsluga.GeneratorOrganizmow;
import projekt.obsluga.Para;

import java.util.Random;
import java.util.Vector;

public class Roslina extends Organizm implements Wypisanie {

    @Override
    public String Wypisz() {
        return null;
    }

    @Override
    public void akcja() {
        if (czyMozeRozmnazac > 10) {
            czyMozeRozmnazac = 0;
            String status;
            Vector<String> kom = swiat.getKomentarz();
            Vector<Para> puste = swiat.znajdzPuste(polozenieX, polozenieY);
            Organizm[][] swiatOrganizmow = swiat.getSwiatOrganizmow();
            Organizm o = new GeneratorOrganizmow().klonowanie(this, swiat);
            if (puste.size() > 0) {
                Random random = new Random();
                int losuj = random.nextInt(puste.size());
                o.setPolozenieX(puste.get(losuj).getL());
                o.setPolozenieY(puste.get(losuj).getR());
                swiatOrganizmow[puste.get(losuj).getL()][puste.get(losuj).getR()] = o;
                swiat.dodajOrganizm(o, true);
                status = Wypisz() + " ROZPYLIL SIE NA POZYCJE " + String.valueOf(puste.get(losuj).getL()) + ":" + String.valueOf(puste.get(losuj).getR());
            } else {
                status = Wypisz() + " NIE ROZPYLIL SIE Z POWODU BRAKU MIEJSCA";
            }
            kom.add(status);
        }
    }

    @Override
    public void kolizja(Organizm atakujacy) {
        String status;
        Vector<String> kom = swiat.getKomentarz();
        Organizm[][] swiatOrganizmow = swiat.getSwiatOrganizmow();
        status = Wypisz() + " ZOSTAL ZJEDZONY PRZEZ " + atakujacy.Wypisz();
        swiatOrganizmow[polozenieX][polozenieY] = atakujacy;
        swiatOrganizmow[atakujacy.getPolozenieX()][atakujacy.getPolozenieY()] = null;
        atakujacy.setPolozenieX(polozenieX);
        atakujacy.setPolozenieY(polozenieY);
        czyZyje = false;
        kom.add(status);
    }
}

File: ObjectProgrammingPython/Rosliny/roslina.py
import random
from Obsluga.generatorOrganizmow import generatorOrganizmow
from Obsluga.para import Para
from organizm import Organizm


class Roslina(Organizm):
    def Wypisz(self):
        pass

    def akcja(self):
        if self.czyMozeRozmnazac > 10:
            self.czyMozeRozmnazac = 0
            status = ""
            kom = self.swiat.getKomentarz()
            puste = self.swiat.znajdzPuste(self.polozenieX, self.polozenieY)
            swiatOrganizmow = self.swiat.getSwiatOrganizmow()
            o = generatorOrganizmow().klonowanie(self, self.swiat)
            if len(puste):
                losuj = random.randint(0,len(puste)-1)
                o.setPolozenieX(puste[losuj].getX())
                o.setPolozenieY(puste[losuj].getY())
                swiatOrganizmow[puste[losuj].getX()][puste[losuj].getY()] = o
                self.swiat.dodajOrganizm(o, True)
                status = self.Wypisz() + " ROZPYLIL SIE NA POZYCJE " + str(puste[losuj].getX()) + ":" + str(puste[losuj].getY())
            else:
                status = self.Wypisz() + " NIE ROZPYLIL SIE Z POWODU BRAKU MIEJSCA"
            kom.append(status)

    def kolizja(self, atakujacy):
        status = ""
        kom = self.swiat.getKomentarz()
        swiatOrganizmow = self.swiat.getSwiatOrganizmow()
        status = self.Wypisz() + " ZOSTAL ZJEDZONY PRZEZ " + atakujacy.Wypisz()
        swiatOrganizmow[self.polozenieX][self.polozenieY] = atakujacy
        swiatOrganizmow[atakujacy.getPolozenieX()][atakujacy.getPolozenieY()] = None
        atakujacy.setPolozenieX(self.polozenieX)
        atakujacy.setPolozenieY(self.polozenieY)
        self.czyZyje = False
        kom.append(status)

